mozetl/search/aggregates.py

```diff
@@ -50,57 +51,85 @@ def agg_first(col):
+
+
 def search_clients_daily(main_summary):
-    return agg_search_data(
-        main_summary,
-        ["client_id", "submission_date", "engine", "source"],
-        list(
-            map(
-                agg_first,
-                [
-                    "country",
-                    "app_version",
-                    "distribution_id",
-                    "locale",
-                    "search_cohort",
-                    "addon_version",
-                    "os",
-                    "channel",
-                    "profile_creation_date",
-                    "default_search_engine",
-                    "default_search_engine_data_load_path",
-                    "default_search_engine_data_submission_url",
-                    "sample_id",
-                ],
+    return (
+        agg_search_data(
+            main_summary,
+            ["client_id", "submission_date", "engine", "source"],
+            list(
+                map(
+                    agg_first,
+                    [
+                        "country",
+                        "app_version",
+                        "distribution_id",
+                        "locale",
+                        "search_cohort",
+                        "addon_version",
+                        "os",
+                        "channel",
+                        "profile_creation_date",
+                        "default_search_engine",
+                        "default_search_engine_data_load_path",
+                        "default_search_engine_data_submission_url",
+                        "sample_id",
+                    ],
+                )
             )
+            + [
+                # Count of 'first' subsessions seen for this client_day
+                (
+                    count(when(col("subsession_counter") == 1, 1)).alias(
+                        "sessions_started_on_this_day"
+                    )
+                ),
+                first(
+                    datediff(
+                        "subsession_start_date",
+                        from_unixtime(col("profile_creation_date") * 24 * 60 * 60),
+                    )
+                ).alias("profile_age_in_days"),
+                sum(col("subsession_length") / 3600.0).alias("subsession_hours_sum"),
+                mean(size("active_addons")).alias("active_addons_count_mean"),
+                (
+                    max(
+                        "scalar_parent_browser_engagement_max_concurrent_tab_count"
+                    ).alias("max_concurrent_tab_count_max")
+                ),
+                (
+                    sum("scalar_parent_browser_engagement_tab_open_event_count").alias(
+                        "tab_open_event_count_sum"
+                    )
+                ),
+                (sum(col("active_ticks") * 5 / 3600.0).alias("active_hours_sum")),
+                (
+                    collect_list("scalar_parent_browser_search_ad_clicks").alias(
```

Comment:
This is causing an error `expression 'search_ad_clicks_sum' cannot be used as a grouping expression because its data type array<map<string,int>> is not an orderable data type.`, @acmiyaguchi any ideas how to work around that? preferrably without rewriting this in spark >_>

Reply:
This is tricky. The pivot aggregation is causing the issues here:
https://github.com/mozilla/python_mozetl/blob/97cb6c243996c2d242a3b427878b19c10e18f177/mozetl/search/aggregates.py#L181-L186

It might be possible to run the aggregation function after the pivot. If not, I think the columns probably need to be exploded in a similar way to `explode_search_counts`.